chainer/functions/math/matmul.py

```diff
@@ -111,6 +112,8 @@ def forward(self, x):
             y = a * b
         else:
             y = _matmul(a, b, self.transa, self.transb, self.transc)
+        if self.dtype is not None and y.dtype != self.dtype:
+            y = y.astype(self.dtype, copy=False)
```

Comment:
This case do not need `copy=False` option.

Reply:
Ah..., thank you for pointing it out! will remote the option.